app/livechat/server/api/v1/room.js

```diff
@@ -40,7 +41,22 @@ API.v1.addRoute('livechat/room', {
 			}
 
 			const rid = roomId || Random.id();
-			const room = Promise.await(getRoom({ guest, rid, agent, extraParams }));
```

Comment:
IMO the logic introduced here in the endpoint scope should be placed inside **getRoom** method rather than here. **getRoom** method is imported in other places too, so logic must be reused.
So, the only change needed would be to pass `departmentId` parameter via **getRoom** method.

Reply:
That could work 🤔 . I'm a bit reluctant to modify the behavior of a function that other places are using, since the new behavior could be not what the callers expect, so I prefered to do that on the endpoint.

Do you have in mind any place that could "break" with the modification? (some place that it's expecting a new room on every call, etc)

Reply:
The `getRoom` method should have the same behaviour everywhere. If the same method is called, the result should be will be same in places.